torch/csrc/api/include/torch/nn/modules/container/functional.h

```diff
@@ -64,16 +64,9 @@ class TORCH_API FunctionalImpl : public torch::nn::Cloneable<FunctionalImpl> {
       typename... Args,
       typename = std::enable_if_t<(sizeof...(Args) > 0)>>
   explicit FunctionalImpl(SomeFunction original_function, Args&&... args)
-      // NOLINTNEXTLINE(modernize-avoid-bind)
-      : function_(std::bind(
-            original_function,
-            /*input=*/std::placeholders::_1,
-            std::forward<Args>(args)...)) {
-    // std::bind is normally evil, but (1) gcc is broken w.r.t. handling
-    // parameter pack expansion in lambdas and (2) moving parameter packs into
-    // a lambda only works with C++14, so std::bind is the more move-aware
-    // solution here.
-  }
+      : function_([=](auto&& input) {
+          return original_function(input, std::forward<Args>(args)...);
```

Comment:
I think you need to move args into the lambda here... Otherwise it;s trying to move the args everytime the function is called
```suggestion
    : function_([original_function = std::forward<SomeFunction>(original_function), ...args = std::forward<Args>(args)](auto&& input) {
        return original_function(std::forward<decltype(input)>(input), std::forward<decltype(args)>(args)...);
    }) {

Reply:
Clang-tidy can handle this conversion for you too I think if remove the NOLINT